### PR TITLE
Pass granularity from backend to frontend as ISO duration

### PR DESCRIPTION
**Pull request by betodealmeida**
Currently the backend will pass the granularity as a string that is engine dependent, e.g, "30 seconds". I changed the backend so that an ISO 8601 duration is passed instead. This allows the frontend to handle the granularity better, eg, for the play slider.

### CODE REVIEW

File: superset/assets/javascripts/explore/stores/controls.jsx [hunk 1]
```diff
@@ -699,21 +699,21 @@ export const controls = {
...
-      '1 day',
-      '7 days',
-      'week',
-      'week_starting_sunday',
-      'week_ending_saturday',
-      'month',
-    ]),
+    choices: [
+      [null, 'all'],
+      ['PT5S', '5 seconds'],
```
Comment (mistercrunch) [inline superset/assets/javascripts/explore/stores/controls.jsx]: If a chart has been saved with a string say `5 seconds` as a value, did you confirm that this load up ok in here as a "free form" value? From my understanding it should.

Comment (mistercrunch) [reply superset/assets/javascripts/explore/stores/controls.jsx]: @betodealmeida ^ please confirm and I'll merge

Comment (betodealmeida) [reply superset/assets/javascripts/explore/stores/controls.jsx]: It does, it loads with the saved value as freeform:

screen shot 2018-04-06 at 11 16 44 am

**Comment by NathanQuinn** [comment]
@betodealmeida @mistercrunch 

I exported a postgres backed set of charts and dashboards to another postgres instance with this change included. It seems to work when viewing the actual chart, but when a time grain is persisted on a chart used in a dashboard I'm getting an error as such:

```
Traceback (most recent call last):
  File "/home/superset/superset/viz.py", line 362, in get_df_payload
    df = self.get_df(query_obj)
  File "/home/superset/superset/viz.py", line 160, in get_df
    self.results = self.datasource.query(query_obj)
  File "/home/superset/superset/connectors/sqla/models.py", line 705, in query
    sql = self.get_query_str(query_obj)
  File "/home/superset/superset/connectors/sqla/models.py", line 419, in get_query_str
    qry = self.get_sqla_query(**query_obj)
  File "/home/superset/superset/connectors/sqla/models.py", line 545, in get_sqla_query
    timestamp = dttm_col.get_timestamp_expression(time_grain)
  File "/home/superset/superset/connectors/sqla/models.py", line 134, in get_timestamp_expression
    expr = grain.function.format(col=expr)
AttributeError: 'str' object has no attribute 'function'
``` 


I was able to backfill values stored on slices.params.time_grain_sqla, to the ISO durations from [here](https://github.com/apache/incubator-superset/blob/426c34ee8643929cdda5a462c6d04289e73cd68b/superset/db_engine_specs.py#L289-L307) and this resolved the issue.  Did I miss a migration someplace or should one be added for this?

**Comment by xrmx** [comment]
@NathanQuinn i think the ".function" part should be removed

**Comment by john-bodley** [comment]
@betodealmeida should there be a DB migration associated with this PR to update slices which reference the deprecated time grains?

cc: @michellethomas

**Pull request merged**